File: 0-frameworks-in-backend/2-csharp/Cat/CatService.cs
// the cat "module" — a dedicated namespace as the bounded context.
namespace FrameworksDemo.Cat;

// record Cat: System.Text.Json camelCases -> {"id":..,"name":..}.
public record Cat(int Id, string Name);

// CatService is created + managed by the ASP.NET IoC container.
public class CatService
{
    private readonly List<Cat> _cats = new() { new(1, "Milo"), new(2, "Luna") };

    // All -> GET /cats.
    public IReadOnlyList<Cat> All() => _cats;

    // SpyHint -> the piece the dog module consumes via DI.
    public string SpyHint() => "cat-network-ready";
}

File: 0-frameworks-in-backend/1-java/src/main/java/com/example/cat/CatService.java
package com.example.cat;

import java.util.List;
import org.springframework.stereotype.Service;

// @Service marks the class for the IoC container (singleton bean by default).
// Equivalent to NestJS's @Injectable.)
@Service
public class CatService {
    private final List<Cat> cats = List.of(new Cat(1, "Milo"), new Cat(2, "Luna"));

    // all() -> GET /cats.
    public List<Cat> all() {
        return cats;
    }

    // spyHint() -> the piece the dog module consumes via DI.
    public String spyHint() {
        return "cat-network-ready";
    }
}
